pkg/config/config.go

```diff
@@ -61,79 +74,81 @@ type Mine struct {
-	MiningInterruptCheckInterval Duration          `json:"miningInterruptCheckInterval"`
-	GasMultiplier                float32           `json:"gasMultiplier"`
-	GasMax                       uint              `json:"gasMax"`
-	NumProcessors                int               `json:"numProcessors"`
-	Heartbeat                    Duration          `json:"heartbeat"`
-	ServerWhitelist              []string          `json:"serverWhitelist"`
-	Worker                       string            `json:"worker"`
-	Password                     string            `json:"password"`
-	PoolURL                      string            `json:"poolURL"`
-	ConfigFolder                 string            `json:"configFolder"`
-	Logger                       map[string]string `json:"logger"`
-	DisputeTimeDelta             Duration          `json:"disputeTimeDelta"` // Ignore data further than this away from the value we are checking.
-	DisputeThreshold             float64           `json:"disputeThreshold"` // Maximum allowed relative difference between observed and submitted value.
 	// Minimum percent of profit when submitting a solution.
 	// For example if the tx cost is 0.01 ETH and current reward is 0.02 ETH
 	// a ProfitThreshold of 200% or more will wait until the reward is increased or
 	// the gas cost is lowered.
 	// a ProfitThreshold of 199% or less will submit
-	ProfitThreshold uint64 `json:"profitThreshold"`
-	// EnvFile location that include all private details like private key etc.
-	EnvFile string `json:"envFile"`
+	ProfitThreshold              uint64
+	Heartbeat                    Duration
+	MiningInterruptCheckInterval Duration
+	MinSubmitPeriod              Duration
+}
+
+type Trackers struct {
+	SleepCycle       Duration
+	FetchTimeout     Duration
+	MinConfidence    float64
+	DisputeTimeDelta Duration // Ignore data further than this away from the value we are checking.
+	DisputeThreshold float64  // Maximum allowed relative difference between observed and submitted value.
+	Names            map[string]bool
 }
 
-const ConfigFolder = "configs"
+// Config holds global config info derived from config.json.
+type Config struct {
+	Mine             Mine
+	DataServer       DataServer
+	Trackers         Trackers          `json:"trackers"`
+	PublicAddress    string            `json:"publicAddress"`
+	EthClientTimeout uint              `json:"ethClientTimeout"`
+	DBFile           string            `json:"dbFile"`
+	GasMultiplier    float32           `json:"gasMultiplier"`
+	GasMax           uint              `json:"gasMax"`
+	ServerWhitelist  []string          `json:"serverWhitelist"`
+	ApiFile          string            `json:"apiFile"`
+	ManualDataFile   string            `json:"manualDataFile"`
+	Logger           map[string]string `json:"logger"`
+	// EnvFile location that include all private details like private key etc.
+	EnvFile      string `json:"envFile"`
+	ConfigFolder string `json:"configFolder"`
```

Comment:
The idea is to remove it completely.

Reply:
yah, soz about that. i had some late night thought about why that would be an issue, but then realised it wasnt and forgot to remove. All gone now.